java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -555,12 +705,21 @@ private boolean handleMethodInvocation(Instruction instruction) {
             }));
         methodInvokedBehavior
           .exceptionalPathYields()
-          .forEach(yield ->
+          .forEach(yield -> {
+            org.sonar.plugins.java.api.semantic.Type exceptionType = yield.exceptionType();
             yield.statesAfterInvocation(
-              Lists.reverse(pop.values), Collections.emptyList(), pop.state, () -> constraintManager.createExceptionalSymbolicValue(yield.exceptionType())).forEach(ps -> {
-              ps.storeExitValue();
-              enqueue(new ProgramPoint(exitBlock), ps);
-            }));
+              Lists.reverse(pop.values), Collections.emptyList(), pop.state, () -> constraintManager.createExceptionalSymbolicValue(exceptionType)
+              ).forEach(ps -> {
+                ps.storeExitValue();
+                programPosition.block.successors().stream()
```

Comment:
I think this deserves to be extracted to the method, it's quite hard to follow, e.g. `List<Block> exceptionHandlers(...)`